src/ValueObjects/FacetResults.php

```diff
@@ -2,115 +2,94 @@
 
 namespace CultuurNet\SearchV3\ValueObjects;
 
+use JMS\Serializer\Annotation\HandlerCallback;
 use JMS\Serializer\DeserializationContext;
 use JMS\Serializer\JsonDeserializationVisitor;
-use JMS\Serializer\TypeParser;
-use JMS\Serializer\Annotation\HandlerCallback;
 
-class FacetResults implements \Iterator
+final class FacetResults implements \Iterator
 {
-
-    protected $key;
+    /**
+     * @var string|null
+     */
+    private $key;
 
     /**
-     * @var array
+     * @var FacetResult[]
      */
-    protected $facetResults;
+    private $facetResults = [];
 
     /**
-     * @return array
+     * @return FacetResult[]
      */
-    public function getFacetResults()
+    public function getFacetResults(): array
     {
         return $this->facetResults;
     }
 
     /**
-     * @param array $facetResults
-     * @return FacetResults
+     * @param FacetResult[] $facetResults
      */
-    public function setFacetResults($facetResults)
+    public function setFacetResults(array $facetResults): void
     {
         $this->facetResults = $facetResults;
-
-        return $this;
     }
 
-    /**
-     * @return array
-     */
-    public function getFacetResultsByField($field)
+    public function getFacetResultsByField($field): array
     {
         $results = [];
         foreach ($this->facetResults as $facetResult) {
-            if ($facetResult->getField() == $field) {
+            if ($facetResult->getField() === $field) {
                 $results[] = $facetResult;
             }
         }
         return $results;
     }
 
-    /**
-     * @inheritDoc
-     */
     public function current()
```

Comment:
Can this get `FacetResult` as a return type?

Reply:
I wasn't sure since it's a method defined on the `\Iterator` interface, but apparently it's possible yes: https://github.com/cultuurnet/search-v3/pull/43/commits/04e2d492eee28bc4f6823c63193a8d1b53bafa8e